app/src/main/java/ar/scacchipa/twittercloneapp/domain/AuthorizationUseCase.kt

```diff
@@ -0,0 +1,45 @@
+package ar.scacchipa.twittercloneapp.domain
+
+import ar.scacchipa.twittercloneapp.datasource.UserAccessToken
+import ar.scacchipa.twittercloneapp.datasource.provideAuthSourceDateApi
+import ar.scacchipa.twittercloneapp.datasource.provideRetrofit
+import ar.scacchipa.twittercloneapp.repository.AuthorizationRepository
+import ar.scacchipa.twittercloneapp.repository.IAuthorizationRepository
+
+class AuthorizationUseCase(
+    private val repository: IAuthorizationRepository =
+        AuthorizationRepository(provideAuthSourceDateApi(provideRetrofit()))
+) {
+    suspend fun generateAccessToken(
+        transitoryToken: String,
+        grant_type: String = "authorization_code",
+        clientId: String,
+        redirect_uri: String,
+        codeVerifier: String = "challenge",
+        state:String = "state"
+    ): UserAccessToken {
+        return repository.genAccessToken(
+            transitoryToken = transitoryToken,
+            grant_type = grant_type,
+            clientId = clientId,
+            redirect_uri = redirect_uri,
+            codeVerifier = codeVerifier,
+            state = state)
+    }
+
+    fun getScope():String  {
+        return listOf("users.read", "tweet.read", "offline.access", "list.read", "follows.read",
+            "like.read", "space.read","tweet.write", "like.write").joinToString("%20")
+    }
+
+    fun createTemporaryCodeUrl(clientId: String, redirectUri: String): String {
```

Comment:
mmm this trigger me some questions. Usually use cases have only one method, usually called invoke() that fires up the hole thing. Having like this second method here makes me wonder if it is in the right place.

Reply:
The use case was splitted into AuthorizationUseCase and ConsumableAuthUseCase.
Both classes have their invoke operators